GSA/COM API/Python/GSA2Compos.py

```diff
@@ -0,0 +1,202 @@
+"""
+Program for linking GSA with Compos for composite beam design
+This script is intended to
+1. open GSA model using gsapy
+2. extract member sections and lengths based on first saved list
+   - write to member_list
+3. extract defined profiles and write to sections_dictionary
+4. write compos_file.csv file using the member_list
+5. open compos_file.csv in Compos, design the members &
+   extract the new sections to compos_results
+6. reopen the GSA model,
+   use compos_results and sections_dictionary to update the members
+"""
+
+import win32com.client  # https://pypi.org/project/pywin32/
+from datetime import datetime
+from tempfile import gettempdir
+from gsapy import GSA
+
+# gsapy references
+# https://gitlab.arup.com/david.dekoning/gsapy/-/blob/master/docs/readme.rst
+# https://gitlab.arup.com/hongkong-rd/JupyterSamples/blob/master/pyGSA/pyGSA_gsapy.ipynb
+
+
+dead_load = 3.0  # kN/m^2
+live_load = 5.0  # kN/m^2
+# add additional variables as necessary to control what is added to Compos
+
+
+def gsa2compos2gsa():
+    global dead_load
+    global live_load
+
+    # get GSA data
+    print("Opening GSA model")
+    gsa_model = GSA(r'C:\Temp\Composite test.gwb', version='10.1')
+
+    # gets the members from the first saved list
+    search_list = \
+        str(gsa_model.get_all_saved_lists()[0]).split("\t")[4].split(" ")
```

Comment:
you can replace your for loop with this:
```python
for i in gsa_model._get_entities_in_named_list("Secondary beams"):  # create list of members to be designed
        member = gsa_model.get_members(i)
        member_list.append([i, gsa_model.get_section_info(member.prop),
                            gsa_model.get_member_length(i)])
```

`model._get_entities_in_named_list` returns a list of integers (member ids). It works even if the list is `1 to 10` or other complex definitions

gsapy creates a Member object with all the info you need, so there is no need to convert to `str` and then parse the string.

`member.prop` is the integer id of the section

Reply:
done